tensorflow/core/kernels/data/map_dataset_op.cc

```diff
@@ -12,219 +12,194 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+#include "tensorflow/core/kernels/data/map_dataset_op.h"
 #include "tensorflow/core/common_runtime/function.h"
 #include "tensorflow/core/framework/dataset.h"
 #include "tensorflow/core/framework/partial_tensor_shape.h"
 #include "tensorflow/core/framework/tensor.h"
-#include "tensorflow/core/kernels/data/captured_function.h"
 #include "tensorflow/core/kernels/data/dataset_utils.h"
+#include "tensorflow/core/kernels/data/name_utils.h"
 #include "tensorflow/core/lib/random/random.h"
 
 namespace tensorflow {
 namespace data {
-namespace {
 
 // See documentation in ../../ops/dataset_ops.cc for a high-level
 // description of the following op.
 
-class MapDatasetOp : public UnaryDatasetOpKernel {
+constexpr char MapDatasetOp::kDatasetType[];
```

Comment:
what do we need this for?

Reply:
@jsimsa Without this, there will be some errors when running the unit test:

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tensorflow/core/kernels/data:map_dataset_op_test
-----------------------------------------------------------------------------
dyld: Symbol not found: __ZN10tensorflow4data12MapDatasetOp12kDatasetTypeE
  Referenced from: /private/var/tmp/_bazel_fei/5dc7b372a7c45427ff30500d3e22fe26/execroot/org_tensorflow/bazel-out/darwin-opt/bin/tensorflow/core/kernels/data/map_dataset_op_test.runfiles/org_tensorflow/tensorflow/core/kernels/data/map_dataset_op_test
  Expected in: flat namespace
 in /private/var/tmp/_bazel_fei/5dc7b372a7c45427ff30500d3e22fe26/execroot/org_tensorflow/bazel-out/darwin-opt/bin/tensorflow/core/kernels/data/map_dataset_op_test.runfiles/org_tensorflow/tensorflow/core/kernels/data/map_dataset_op_test
external/bazel_tools/tools/test/test-setup.sh: line 310: 19316 Abort trap: 6           "${TEST_PATH}" "$@" 2>&1
```